build_runner/lib/src/server/graph_viz_main.dart

```diff
@@ -15,21 +16,42 @@ main() async {
   searchForm.onSubmit.listen((e) async {
     e.preventDefault();
     var query = searchBox.value;
-    _focus(query);
+    await _focus(query);
     return null;
   });
   _graphReference.callMethod('initializeGraph', [_focus]);
 }
 
-void _focus(String query) async {
-  var parts = query.split('|');
-  var package = parts[0];
-  var path = parts[1];
-  var requestPath = path.startsWith('lib/')
-      ? 'packages/$package/${path.substring(4)}'
-      : '${path.split('/').skip(1).join('/')}';
-  var nodeInfo =
-      json.decode(await HttpRequest.getString('/\$graph/$requestPath'));
+void _error(String message, Object error, StackTrace stack) {
+  // TODO: update the user page some how?
+  window.console.error(message);
+  window.console.log(error);
+  window.console.log(stack);
+}
+
+Future _focus(String query) async {
+  String requestPath;
+  try {
+    var parts = query.split('|');
+    var package = parts[0];
+    var path = parts[1];
+    requestPath = path.startsWith('lib/')
+        ? 'packages/$package/${path.substring(4)}'
+        : '${path.split('/').skip(1).join('/')}';
+  } catch (e, stack) {
+    _error('The query you provided ($query) could not be parsed.', e, stack);
```

Comment:
What do the queries look like here? Should we wrap it in back tics?

Reply:
It's just an asset ID – I think. The parens should be fine.